### DISCUSSION TITLE
Swapping AnimatedSprite

### DISCUSSION THREAD

**Discussion by mcherneski** [pixijs/pixijs#9131] [Q&A] [answered]
Hello all, 

I'm exploring Pixi to make a game but I'm having issues with my AnimatedSprite. Currently all this code is running in a single scene but I'm hoping to break it into a character class. 

Right now I'm tinkering with a single character. I preloaded sprite sheets according to a recipe on PixiJS Elementals (shoutout to that amazing site). I then create a map and populate it with AnimatedSprites via a for loop and const animationData = AnimatedSprite.fromFrames(). Code below. 
```

 const animations = Assets.cache.get('duck').data.animations;
        const animationMap = new Map();
        for (let animation in animations)
        {
            const animationData = AnimatedSprite.fromFrames(animations[animation]);
            animationMap.set(animation, animationData)
        }
        this.spriteAnimationMap = animationMap;

```

All of that works well enough. I can call the map by the animation name and retreive the animationData, which in this case is just an AnimatedSprite. My issue is updating the current animated sprite on a trigger. I currently have class member, this.plappy which equals the default AnimatedSprite I want when the game loads. I set this in my constructor. I have a function, changeAnimation, which is supposed to take the name of an animation and the animation map as parameters, and set this.plappy to the new AnimatedSprite.

```
  public changeAnimation(animationName: string, animationMap: any): void {
        const animation = animationMap.get(animationName);
        console.log(animation);
        this.plappy = animation;
        this.plappy.play();
    }
```

When the character reaches the end of the screen, it should call this.changeAnimation('Idle', spritemap) and set the animation to the idle animation. However, when it gets to the end of the screen, the character stops (expected) but the AnimatedSprite doesn't change animations, it just moonwalks in place. 

```
        if (this.plappy.x > Manager.width - 300) {
            this.plappyVelocity = 0;
            this.changeAnimation('Idle', this.spriteAnimationMap);

        }
```
I've seen plenty of examples of swapping regular sprites but I can't find anything on swapping animated sprites. I'm not sure what else to try. Most of the properties in the AnimatedSprite object are read only so I can't swap the textures. Do most folks use the Sprite class rather than AnimatedSprite? What am I missing here? I can send over my full code if that would help folks understand the issue better. 

(Also if you think my mapping idea is not good, I'm more than open to ideas.)

I appreciate any help!

**Comment by miltoncandelero** [comment, marked as answer]
Each animated sprite in your map is added to the stage. Calling them to play isn't enough. You need to make the Animated sprite you want to show visible and the rest invisible. You can use the `visible` property or `removeChild` and `addChild` methods

**Reply by mcherneski** [reply]
I can't believe I missed that! Thank you so much for the assistance.

**Comment by RickeyWard** [comment]
you don't have to make an animated sprite per animation, you can just change the animation by changing the "textures" property on an animated sprite to a different animation in the spritesheet

```
animatedSprite.textures = aniSpriteSheet.animations["nameofanimation"];
// be sure to call play again.
```
